tools/torch/main.lua

```diff
@@ -485,11 +485,13 @@ if opt.optimState ~= '' then
 end
 
 local function updateConfusion(y,yt)
-    if confusion ~= nil then
-        confusion:batchAdd(y,yt)
+    if trainConfusion ~= nil then
+        trainConfusion:batchAdd(y,yt)
     end
 end
 
+local label_function = network.labelHook or function (input, dblabel) return dblabel end
```

Comment:
I did not get  which is the consistent naming convention in this file because it was mixed inside the same Validation() function and in global variables. For example, the following names are used in the code: **network_func**, **epoch_round**, **Weights**, **Gradients**, **trainBatchSize**, etc.

I could make them consistent if only I knew which notation to use.